### DISCUSSION TITLE
GriefDefender converter

### DISCUSSION THREAD

**Discussion by RoboMWM** [GriefPrevention/GriefPrevention#2248] [Tools]
[See HorusApl's comment below for instructions and download.](https://github.com/GriefPrevention/GriefPrevention/discussions/2248#discussioncomment-8737983)

**Comment by HorusApl** [comment]
Steps

1.) Unzip GDConverter.zip
2.) Navigate to plugins/GriefDefender/worlds/<world_type>/<world_name>/ClaimData/basic
3.) Copy all claim files in this folder, and in the subdivision folder. You can do this for one world at a time,
but or all worlds at once. Script has been tested on 7k~ claims at once with no errors. See notes at bottom.
4.) Move copied files to the folder titled GriefDefenderFiles inside the GDConverter
5.) Open the "worldNames.py" with a text editor of your choice
6.) Change the "<griefdefender_world_UUID>" to the world-UUID found inside the claim data file we copied on
step 4. Change the world name according to what that UUID was if you're using non-standard world names.
Example provided in the file itself.
7.) Save changes after you're done with all your world names/uuids that you have
8.) Double-click the files in this order: worldNames.py, findWorlds.py and then wait for any black screen to
close itself. When it does, double click coverter.py. Depending on how many claims you have, this could take
seconds or minutes. With 7k claims, it took about 45 seconds.
9.) Your newly converted GriefPrevention files will be located in the folder "converted" which you can now copy
directly into this file path: plugins/GriefPreventionData/ClaimData
10.) **IMPORTANT** Open the file named _nextClaimID and input the number of claims you migrated plus
one. If you moved 6969 files, you need to set the number inside that file to 6970.
11.) Done.

GriefDefender uses a Wilderness claim you can set flags for specifically, do not convert that one as this wasn't
intended for that. Use WorldGuard for that, not this tool.

GriefPrevention as of version 16.18.2 doesn't support 3D claims but GriefDefender does. The tool will make all

converted 3D claims into sky to bedrock claims like all other GriefPrevention claims.

If you're doing subclaims, they must be done at the same time as the main claims. Do not do them separately.****

Adding a second copy to make sure the latest one was posted.
[GDConverter.zip](https://github.com/GriefPrevention/GriefPrevention/files/14551688/GDConverter.zip)
